### PR TITLE
Fixed docs in spearmanr

### PR DESCRIPTION
**Pull request by Sebascn**
Fixed the description of axis argument.

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -2896,9 +2896,9 @@ def spearmanr(a, b=None, axis=0):
         `axis`. Both arrays need to have the same length in the `axis`
         dimension.
     axis : int or None, optional
-        If axis=0 (default), then each column represents a variable, with
-        observations in the rows. If axis=0, the relationship is transposed:
-        each row represents a variable, while the columns contain observations.
+        If axis=0 (default), then each row represents a variable, with
+        observations in the columns. If axis=1, the relationship is transposed:
+        each column represents a variable, while the rows contain observations.
         If axis=None, then both arrays will be raveled.
```
Comment (larsmans) [inline scipy/stats/stats.py]: I don't think this is correct. The default behavior does treat each column as a variable. Check the example given below this paragraph:

```py
>>> x2n=np.random.randn(100, 2)
>>> spearmanr(x2n)
(0.022106210621062106, 0.82718799420247646)
```

Two columns, two variables, one output. For three columns, we get a 3×3 output, as expected.

The only mistake is that the second option is also described as `axis=0`.

Comment (Sebascn) [inline scipy/stats/stats.py]: Are you sure? You may be right, your code does make for some valid arguments, but try stating the axis. When no axis are stated then they are raveled. I am not entirely sure what that means though. I was showing in an issue a few days ago why I believe in correctness of my fix. Try look it up in clised issues.

Comment (larsmans) [inline scipy/stats/stats.py]: When no axis is given, it defaults to 0. Check the code.

**Comment by larsmans** [comment]
Found the issue, it's #4965. The thing is that `axis=1` makes no sense when the inputs are both 1-d. The docs might be changed to reflect that, although personally I wouldn't expect `axis=1` to make sense when there is only a dimension 0.

**Comment by larsmans** [comment]
Correction: when the inputs are 1-d, the behavior is actually different. Since 1-d arrays are considered rows by default, the following is inconsistent:

```py
>>> x2n = np.random.randn(10, 2)
>>> rho2d = spearmanr(x2n)[0]
>>> rho1d = spearmanr(x2n[:, 0], x2n[:, 1])[0]
>>> rho1d == rho2d
True
>>> x2n[:, 0].shape
(10,)
```

So the docs for `axis` should reflect that if the inputs are 1-d, observations are in the columns and the `axis` argument should be left alone.

**Pull request closed without merging**